src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,199 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.util.ArrayList;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * Provide a list that all the data can be stored into.
+ * Support functions including add, delete, print tasks and find keywords of a task.
+ */
+public class TaskList {
+    private static ArrayList<Task> tasksList;
+
+    /**
+     * A constructor that initialize the taskList when there is no existing taskList.
+     */
+    public TaskList(){
+        this.tasksList = new ArrayList<>();
+    }
+
+    /**
+     * A constructor that initialize the taskList when there is an existing taskList.
+     *
+     * @param tasksList the existing taskList that we want our tasksList to be equal to.
+     */
+    public TaskList(ArrayList<Task> tasksList){
+        this.tasksList = tasksList;
+    }
+
+    /**
+     * Add a task to the taskList.
+     *
+     * @param task the task to be added.
+     */
+    public void add(Task task){
+        tasksList.add(task);
+    }
+
+    /**
+     * remove the task from the taskList.
+     *
+     * @param index the index of the removed task.
+     * @return the removed task.
+     */
+    public Task remove(int index){
+        Task removeTask = tasksList.remove(index);
+        return removeTask;
+    }
+
+    /**
+     * get the task with this index.
+     *
+     * @param index the index of the needed task.
+     * @return the task with this index.
+     */
+    public Task get(int index){
+        Task getTask=tasksList.get(index);
+        return getTask;
+    }
+
+    /**
+     * get the whole taskList.
+     *
+     * @return the corresponding taskList.
+     */
+    public ArrayList<Task> getTasksList(){
+        return tasksList;
+    }
+
+    /**
+     * Print everything in the taskList.
+     */
+    public static void printList() {
+        int count = 1;
+        for(Task task : tasksList){
+            System.out.print(count + ". ");
+            System.out.println(task);
+            count++;
+        }
+    }
+
+    /**
+     * To find corresponding tasks which include the required keyword.
+     *
+     * @param inputLine user input command to show the required keyword.
+     */
+    public static void findTasks(String inputLine) {
+        String findLine = inputLine.substring(5);
```

Comment:
maybe magic number here? please refer to: https://nus-cs2113-ay2021s1.github.io/website/se-book-adapted/chapters/codeQuality.html#guideline-name-well